aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -20,9 +20,58 @@ bool allclose(const Tensor& self, const Tensor& other, double rtol, double atol,
 
 Tensor isclose(const Tensor& self, const Tensor& other, double rtol, double atol, bool equal_nan) {
   // TODO: use bitwise operator overloads once we add them
-
   TORCH_CHECK(self.scalar_type() == other.scalar_type(), self.scalar_type(), " did not match ", other.scalar_type())
 
+  // Checks that rtol and atol are non-negative
+  // Note: consistent with Python
+  TORCH_CHECK(rtol >= 0, "rtol must be greater than or equal to zero, but got ", rtol);
+  TORCH_CHECK(atol >= 0, "atol must be greater than or equal to zero, but got ", atol);
+
+  // Implements isclose for complex tensors, consistent with
+  // Python's cmath.isclose().
```

Comment:
I have a slight edit here that clarifies we're consistent with NumPy but our non-finite value behavior is derived from cmath.isclose.